agent/config/config.go

```diff
@@ -399,6 +419,7 @@ type ServiceDefinition struct {
 	TaggedAddresses   map[string]ServiceAddress `mapstructure:"tagged_addresses"`
 	Meta              map[string]string         `mapstructure:"meta"`
 	Port              *int                      `mapstructure:"port"`
+	Ports             ServicePorts              `mapstructure:"ports"`
 	SocketPath        *string                   `mapstructure:"socket_path"`
 	Check             *CheckDefinition          `mapstructure:"check"`
 	Checks            []CheckDefinition         `mapstructure:"checks"`
```

Comment:
```suggestion
	Checks            []*CheckDefinition         `mapstructure:"checks"`
```

Reply:
@srahul3  This is existing code. Are you sure to change this?